group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1967,6 +2523,60 @@ private Assignment updateTargetAssignment(
         }
     }
 
+    /**
+     * Updates the target assignment according to the updated member and subscription metadata.
+     *
+     * @param group                 The ShareGroup.
+     * @param groupEpoch            The group epoch.
+     * @param updatedMember         The updated member.
+     * @param subscriptionMetadata  The subscription metadata.
+     * @param subscriptionType      The group subscription type.
+     * @param records               The list to accumulate any new records.
+     * @return The new target assignment.
+     */
+    private Assignment updateTargetAssignment(
```

Comment:
This one looks very similar to the existing one. Would it be possible to reuse it? We could perhaps parameterize the type of the member. I am not sure whether it works or not.

Reply:
Though it looks similar but I felt it will be cleaner to have separate methods as the operations and construction of TargetAssignmentBuilder varies.